lib/similarity_utils.py

```diff
@@ -121,6 +121,14 @@ def print_top_similiar(main_actor: str, names: List[str], values: List[float]) -
 def get_ranking(data: pd.DataFrame, main_actor_id: str, ranking_length: int = 5) -> List[str]:
     """metoda dla ramki danych, id aktora oraz (opcjonalnie) długości rankingu, zwraca listę id aktorów najbardziej
     podobnych do wybranego aktora"""
+    warnings.warn("Jest nowa funkcja get_ranking_for_actor_data_row")
     main_actor = prepare_pandas_row(find_actor(data, main_actor_id))
     ids, similarities = similarity_one_vs_all(data, main_actor)
     return select_top_similiar(ids, similarities, ranking_length)[0]
+
+def get_ranking_for_actor_data_row(ranking_data_source: pd.DataFrame, actor_data_row: pd.DataFrame, ranking_length: int = 5) -> List[str]:
+    """metoda dla ramki danych, wiersza wybranego aktora oraz (opcjonalnie) długości rankingu, zwraca listę id aktorów najbardziej
+    podobnych do wybranego aktora"""
+    actor_pandas_row = prepare_pandas_row(actor_data_row)
+    ids, similarities = similarity_one_vs_all(ranking_data_source, actor_pandas_row)
+    return select_top_similiar(ids, similarities, ranking_length)[0]
```

Comment:
Weszły zmiany na maina, w których dodana została nowa wersja liczenia similarity oraz opcjonalny parametr dla m.in. metod `similarity_one_vs_all` oraz `similarity_one_vs_all_new`.  Spójrz proszę na zmiany i zaadoptuj do nich swoje rankingi oraz (jeżeli potrzebujesz) to stwórz metodę rangingującą używającą metodę `similarity_one_vs_all_new`.